### PR TITLE
EAHW-2518: Batch update accruals

### PR DESCRIPTION
**Pull request by joannafawl**
Added `patchAccruals` function to restClient which generates the patch payload body and calls patch endpoint

### CODE REVIEW

File: src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/TimeEntryConsumerTest.java [hunk 1]
```diff
@@ -44,7 +49,13 @@
 class TimeEntryConsumerTest {
 
   @Captor
-  private ArgumentCaptor<TimeEntry> timeEntryCaptor;
+  private ArgumentCaptor<TimeEntry> timeEntryCaptor1;
```
Comment (taomoh) [inline src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/TimeEntryConsumerTest.java:52]: Does this variable name need to have `1` in it?

File: src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java [hunk 2]
```diff
@@ -96,4 +106,33 @@ public Agreement getApplicableAgreement(String tenantId, String personId, LocalD
...
+      PatchBody blob = new PatchBody();
+
+      blob.setOp("replace");
+      blob.setPath("/" + a.getId().toString());
+      blob.setValue(a);
+
+      body.add(blob);
+    });
+
+    return body;
```
Comment (taomoh) [inline src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java:136]: You may have a different preference. but using stream would shave of a few lines of code:

```
    return accruals.stream()
        .map(a -> {
          PatchBody blob = new PatchBody();

          blob.setOp("replace");
          blob.setPath("/" + a.getId().toString());
          blob.setValue(a);

          return blob;
        })
        .toList();
```

**Review by taomoh — APPROVED** [review]
Minor optional comments

File: src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/TimeEntryConsumerTest.java (hunk 1 above)
Comment (joannafawl) [reply src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/TimeEntryConsumerTest.java:52]: Good catch, i was meant to remove that

File: src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java (hunk 2 above)
Comment (joannafawl) [reply src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java]: Updated

**Review by sewerynmi — DISMISSED** [review]

**Review by taomoh — APPROVED** [review]

**Review by taomoh — APPROVED** [review]

**Review by JWeston-HO — APPROVED** [review]

**Pull request merged**